lib/ui/mail_view_indv.dart

```diff
@@ -40,23 +41,42 @@ class MailPieceViewWidgetState extends State<MailPieceViewWidget> {
   late String originalText = widget.mailPiece.imageText;
   String mailPieceText = '';
 
-  late bool hasLearnMore = false;
-  late Uri learnMoreLinkUrl = Uri.parse("https://www.google.com");
-  late Uri reminderLinkUrl = Uri.parse("https://www.google.com");
-
+  bool loading = true;
 
-  //these Html links really aren't used - delete eventually.  URL launcher works better
-  late String learnMoreLinkHtml = '';
-  late String reminderLinkHtml = '';
+  late bool hasLearnMore = false;
+  late bool hasDoMore = false;
+  late Uri? learnMoreLinkUrl = null;
+  late Uri? reminderLinkUrl = null;
 
 
   MailPieceViewWidgetState();
 
   @override
   void initState() {
     super.initState();
-    _getMailPieceEmail();
-    mailPieceText = _reformatMailPieceString(originalText);
+    setState(() {
+      loading = true;
+    });
+    if(widget.mailPiece.scanImgCID != "") {
+      _getMailPieceEmail();
+      mailPieceText = _reformatMailPieceString(originalText);
+    } else {
+      // scanImgCID = "" means no image or links will be found. dont check email and remove do more section
+      setState(() {
+        mailImage = null;
+        reminderLinkUrl = null;
+        learnMoreLinkUrl = null;
+        hasDoMore = false;
+        hasLearnMore = false;
+        loading = false;
+      });
+    }
+    /*
+    setState(() {
+      loading = false;
```

Comment:
Please remove all commented out code (unless we will need it later)

Reply:
removed